test/test_fx.py

```diff
@@ -563,6 +563,165 @@ def test_construct_root_dict(self):
+            # orig_base_module: original torch.nn.module which orig_nodes's torch.fx.GraphModule was traced from
+            # submodule_hints: if adding submodules from modules created by this function, add their names here
+            #                  and add their names to module_node_names and add their nodes to orig_nodes
+            def create_module_from_nodes(
+                module_name: str,
+                module_node_names,
+                orig_nodes: Dict[str, torch.fx.Node],
+                orig_base_module: torch.nn.Module,
+                submodule_hints=[]
+            ) -> (torch.fx.GraphModule, torch.fx.Node):
+                # create new graph
+                new_nodes: Dict[str, torch.fx.Node] = {}
+                graph = torch.fx.Graph()
+                graph_placeholders = (
+                    []
+                )  # used to keep track of placeholders to make torch.fx.Node out of created module
+
+                # copy over nodes from original graph
+                for node_name in module_node_names:
+                    orig_node = orig_nodes[node_name]
+
+                    # perform arg transformation
+                    new_args = []
+                    for orig_arg in orig_node.args:
+                        if isinstance(orig_arg, torch.fx.Node):
+                            # TODO assumption that nodes aren't nested in lists/tuples
+                            new_args.append(new_nodes[orig_arg.name])
+                        else:
+                            # TODO assumption that non-nodes are directly copyable
+                            new_args.append(orig_arg)
+                    new_args = tuple(new_args)
+
+                    # perform kwarg tranformation
+                    new_kwargs: Dict[str, torch.fx.Argument] = {}
+                    for key, value in orig_node.kwargs.items():
+                        if isinstance(value, torch.fx.Node):
+                            new_kwargs[key] = new_nodes[value.name]
+                        else:
+                            new_kwargs[key] = value
+
+                    # create target
+                    target = (
+                        module_name + "." + orig_node.name
+                        if orig_node.op in ["get_param", "call_module"]
+                        else orig_node.target
+                    )
+
+                    # create new node
+                    new_nodes[node_name] = graph.create_node(
+                        op=orig_node.op,
+                        target=target,
+                        args=new_args,
+                        kwargs=new_kwargs,
+                        name=orig_node.name,
+                    )
+                    if node_name == module_node_names[-1]:
+                        graph.output(new_nodes[node_name])
+
+                    if orig_node.op == "placeholder":
+                        graph_placeholders.append(new_nodes[node_name])
```

Comment:
For the Graph inputs and outputs here, you'll want to determine which data dependencies come from outside the subgraph (inputs) and which are used outside the subgraph. I think I can explain better offline